scripts/build_test.py

```diff
@@ -721,4 +722,74 @@ def mock_require_compiled_js_dir_to_be_valid():
                 self.assertFalse(
                     os.path.exists(os.path.dirname(MOCK_COMPILED_JS_DIR)))
 
+    def test_build_with_prod_env(self):
+        def mock_build_using_webpack():
+            pass
+
+        def mock_ensure_files_exist(unused_filepaths):
+            pass
+
+        def mock_compile_typescript_files(unused_project_dir):
+            pass
+
+        ensure_files_exist_swap = self.swap(
+            build, '_ensure_files_exist', mock_ensure_files_exist)
+        build_using_webpack_swap = self.swap(
+            build, 'build_using_webpack', mock_build_using_webpack)
+        compile_typescript_files_swap = self.swap(
+            build, 'compile_typescript_files', mock_compile_typescript_files)
+        args_swap = self.swap(sys, 'argv', ['build.py', '--prod_env'])
+
+        with ensure_files_exist_swap, build_using_webpack_swap, (
+            compile_typescript_files_swap), args_swap:
+            build.build()
```

Comment:
No assertions? that doesn't seem right?

Reply:
Hmm I am actually testing the build() method [here](https://github.com/oppia/oppia/blob/2858751c4bc680d1f1ff97af2f917ad162763156/scripts/build.py#L1298) by mocking most of the build functions. So actually these are not testing anything really but just making sure the functions are called.

Reply:
Can you please elaborate this? How do we ensure that the correct functions are only called?

Reply:
Sorry my mistake -- we are not ensuring if the correct functions are called or not, I don't think we can do that with our current codebase. As I mentioned this is not really testing anything as the important functions are swapped, its just calling build(). The functions are already tested separately before.

Reply:
Rishav, now, I am a bit confused here -- what are we actually testing? If we're just calling build without testing anything, then why do we need this? Thanks!

Reply:
We are just testing [this](https://github.com/oppia/oppia/blob/2858751c4bc680d1f1ff97af2f917ad162763156/scripts/build.py#L1298) method, i.e we need to call this method somehow to increase the coverage. The actual functions are tested already.

Reply:
Hmmm, I am not sure about this. Any ideas here @brianrodri? Other than this, LGTM.

Reply:
Made some changes after discussing with @brianrodri